src/System.Linq/src/System/Linq/Last.cs

```diff
@@ -109,28 +147,46 @@ public static TSource LastOrDefault<TSource>(this IEnumerable<TSource> source)
                         do
                         {
                             result = e.Current;
-                        } while (e.MoveNext());
+                        }
+                        while (e.MoveNext());
```

Comment:
If we have black lines after each block, should we have blank lines after do…while?